modules/move-etcd-different-disk.adoc

```diff
@@ -0,0 +1,182 @@
+          ExecCondition=/usr/bin/test ! -d /var/lib/etcd/member
+          ExecStart=/usr/sbin/setenforce 0
+          ExecStart=/bin/rsync -ar /sysroot/ostree/deploy/rhcos/var/lib/etcd/ /var/lib/etcd/
+          ExecStart=/usr/sbin/setenforce 1
+          TimeoutSec=0
+
+          [Install]
+          WantedBy=multi-user.target graphical.target
+        enabled: true
+        name: sync-var-lib-etcd-to-etcd.service
+      - contents: |
+          [Unit]
+          Description=Restore recursive SELinux security contexts
+          DefaultDependencies=no
+          After=var-lib-etcd.mount
+          Before=crio.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecStart=/sbin/restorecon -R /var/lib/etcd/
+          TimeoutSec=0
+
+          [Install]
+          WantedBy=multi-user.target graphical.target
+        enabled: true
+        name: restorecon-var-lib-etcd.service
+
+----
+
+.  Create the machine configuration by entering the following commands:
++
+[source,terminal]
+----
+$ oc login -u ${ADMIN} -p ${ADMINPASSWORD} ${API}
+... output omitted ...
+----
++
+[source,terminal]
+----
+$ oc create -f etcd-mc.yml
+machineconfig.machineconfiguration.openshift.io/98-var-lib-etcd created
+----
++
+[source,terminal]
+----
+$ oc login -u ${ADMIN} -p ${ADMINPASSWORD} ${API} [... output omitted ...] $ oc create -f etcd-mc.yml machineconfig.machineconfiguration.openshift.io/98-var-lib-etcd created
+----
++
+The nodes update and reboot. After the reboot completes, the following events occur:
++
+*  An XFS filesystem is created on the specified disk.
+*  The disk mounts to `/var/lib/etc`.
+*  The content from `/sysroot/ostree/deploy/rhcos/var/lib/etcd` syncs to `/var/lib/etcd`.
+*  A restore of `SELinux` labels is forced for `/var/lib/etcd`.
+*  The old content is not removed.
+.  After the nodes are on a separate disk, update the machine configuration file, `etcd-mc.yml` with the following information:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 98-var-lib-etcd
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - contents: |
+          [Unit]
+          Description=Mount /dev/sdb to /var/lib/etcd
+          Before=local-fs.target
+          Requires=systemd-mkfs@dev-sdb.service
+          After=systemd-mkfs@dev-sdb.service var.mount
+
+          [Mount]
+          What=/dev/sdb
+          Where=/var/lib/etcd
+          Type=xfs
+          Options=defaults,prjquota
+
+          [Install]
+          WantedBy=local-fs.target
+        enabled: true
+        name: var-lib-etcd.mount
+----
+. Apply the modified version that removes the logic for creating and syncing the device by entering the following command:
++
+[source,terminal]
+----
+$ oc replace -f etcd-mc.yml
+----
++
+The previous step prevents the nodes from rebooting.
```

Comment:
@tmalove what means of "The previous step prevents the nodes from rebooting."? which previous step? thanks

Reply:
I suppose it's "oc replace" step, exact?